### PR TITLE
[LLDB][Windows] Fix watchpoints for Windows

### PR DESCRIPTION
**Pull request by AlexK0**
Hello!

Currently, watchpoints don't work on Windows (this can be reproduced with the existing tests). This patch fixes the related issues so that the tests and watchpoints start working.

Here is the list of tests that are fixed by this patch (on Windows, checked in **release/18.x** branch):
- commands/watchpoints/hello_watchpoint/TestMyFirstWatchpoint.py
- commands/watchpoints/multiple_hits/TestMultipleHits.py
- commands/watchpoints/multiple_threads/TestWatchpointMultipleThreads.py
- commands/watchpoints/step_over_watchpoint/TestStepOverWatchpoint.py
- commands/watchpoints/unaligned-watchpoint/TestUnalignedWatchpoint.py
- commands/watchpoints/watchpoint_commands/TestWatchpointCommands.py
- commands/watchpoints/watchpoint_commands/command/TestWatchpointCommandLLDB.py
- commands/watchpoints/watchpoint_commands/command/TestWatchpointCommandPython.py
- commands/watchpoints/watchpoint_commands/condition/TestWatchpointConditionCmd.py
- commands/watchpoints/watchpoint_count/TestWatchpointCount.py
- commands/watchpoints/watchpoint_disable/TestWatchpointDisable.py
- commands/watchpoints/watchpoint_size/TestWatchpointSizes.py
- python_api/watchpoint/TestSetWatchpoint.py
- python_api/watchpoint/TestWatchpointIgnoreCount.py
- python_api/watchpoint/TestWatchpointIter.py
- python_api/watchpoint/condition/TestWatchpointConditionAPI.py
- python_api/watchpoint/watchlocation/TestTargetWatchAddress.py

### CODE REVIEW

File: lldb/source/Plugins/Process/Windows/Common/ProcessWindows.cpp [hunk 1]
```diff
@@ -406,7 +406,7 @@ void ProcessWindows::RefreshStateAfterStop() {
                m_session_data->m_debugger->GetProcess().GetProcessId(), pc, id);
 
       stop_info = StopInfo::CreateStopReasonWithWatchpointID(
-          *stop_thread, id, m_watchpoints[id].address);
+          *stop_thread, id);
```
Comment (jasonmolenda) [inline lldb/source/Plugins/Process/Windows/Common/ProcessWindows.cpp:409]: Ah yikes, I believe I broke this in https://reviews.llvm.org/D147816 and that `addr_t` is being treated like a boolean now :(

**Review by jasonmolenda — APPROVED** [review]
Thanks for the fixes, these are correct and they look like they're caused by me when I was doing my work on watchpoints earlier this year.  Do you have permissions to merge the PR?

**Comment by AlexK0** [comment]
> Do you have permissions to merge the PR?

I don't have the necessary permissions. Could you please merge the PR when it becomes possible?

**Comment by DavidSpickett** [comment]
First, thanks for fixing this. It has been on Linaro's backlog for a while but as usual resources are scarce.

> Currently, watchpoints don't work on Windows (this can be reproduced with the existing tests).

For background here, the Windows LLDB bot is run by Linaro and is Windows on Arm. As the code for Windows is trying to use Intel parts of the register context, that bot skips the watchpoint test category completely.

(I think we fail to detect any hardware code breakpoints, so those tests get skipped transparently)

**Pull request merged**